### DISCUSSION TITLE
Installation on Fedora

### DISCUSSION THREAD

**Discussion by jcoutch** [DistroAV/DistroAV#646] [General]
I was able to install the obs-ndi plugin on Fedora 30 using the existing Debian packages, and wanted to document the steps that I took, and a small workaround I had to do:

```bash
# Install dependencies
sudo dnf install -y avahi
sudo systemctl enable --now avahi-daemon.service

# Install obs-ndi
sudo dnf install -y dpkg
sudo dpkg -i libndi3_3.8.0-1_amd64.deb
sudo dpkg -i obs-ndi_4.6.0-1_amd64.deb

# The obs-ndi package drops the plugin in the wrong location (note `lib64` vs `lib`)
# I'm creating a symlink as a workaround
sudo ln -s /usr/lib/obs-plugins/obs-ndi.so /usr/lib64/obs-plugins/obs-ndi.so
```

After a reboot, the plugin showed up in OBS and works great!  The only issue was figuring out the actual plugin folder location.

**Comment by gbraad** [comment]
A few weeks ago I did the same, but I never enabled avahi...

**Comment by gbraad** [comment]
There is a pull request for an RPM spec #299, but it is incomplete/lacks requirements.

**Comment by gbraad** [comment]
Duplicate of #137

**Comment by jcoutch** [comment]
Avahi is a dependency of `ndi-sdk` (since it uses mDNS under the covers), and needs to be enabled.  AFAIK, it's enabled by default on most distributions...but wasn't in my case.

**Comment by gbraad** [comment]
I noticed it activated it again using the `avahi-daemon.socket`. [Ref](https://support.newtek.com/hc/en-us/articles/218109477-NDI-Discovery-and-Registration)

**Comment by gbraad** [comment]
Lets try to get this fixed with an actual packages: see #299

**Comment by taspts** [comment]
> I was able to install the obs-ndi plugin on Fedora 30 using the existing Debian packages, and wanted to document the steps that I took, and a small workaround I had to do:
> 
> sudo dpkg -i libndi3_3.8.0-1_amd64.deb
> sudo dpkg -i obs-ndi_4.6.0-1_amd64.deb

dpkg -i  
wants obs-studio and don't recognice that obs-studio is installed by rpm on fedora 32

**Comment by eelcohn** [comment]
@jcoutch Thank you for these instructions!

@taspts this is because obs-studio is installed by the `dnf` package manager, and not by the `dpkg` package manager. Try the following:
```
sudo dpkg --ignore-depends=obs-studio -i ./obs-ndi_4.9.1-1_amd64.deb
```

**Comment by kuboosoft** [comment]
rpm available for Fedora 30-34... :smiley_cat: 
https://unitedrpms.sourceforge.io/32/x86_64/repoview/obs-ndi.html
https://unitedrpms.github.io/

**Comment by brijeshdhaker** [comment]
It also work smoothly on fedora - 34  ...

**Comment by Winor** [comment]
Did anyone successfully installed DistroAV 6 on fedora 41?

**Reply by RayneYoruka** [reply]
Someone does pack an unnoficcial version or the plugin. It's within rpmfusion and we don't mantain it ourselves. I don't even know if they changed from the old obs-ndi. Last time I used it for testing was in Fedora 40.
